btree/btree.c

```diff
@@ -2921,6 +2925,493 @@ static void ff_capture(struct slot *slot, struct m0_be_tx *tx)
 	M0_BTREE_TX_CAPTURE(tx, seg, h, hsize);
 }
 
+/**
+ *  --------------------------------------------
+ *  Section END - Fixed Format Node Structure
+ *  --------------------------------------------
+ */
+
+/**
+ *  --------------------------------------------
+ *  Section START -
+ *  Variable Sized Key and Values Node Structure
+ *  --------------------------------------------
+ *
+ * Proposed node structure
+ *
+ *
+ *                        +--------------------------+
+ *                        |                          |
+ *                 +------+-----------------------+  |
+ *                 |      |                       |  |
+ *             +---+------+-------------------+   |  |
+ *             |   |      |                   |   |  | DIR                    16   11       3  0
+ * +----------+v-+-v--+---v----+-------------++-+-++-++--+--+--+--------------+----+--------+--+
+ * |          |  |    |        |             |  |  |  |  |  |  |              |    |        |  |
+ * | Node Hdr |K0| K1 |   K3   |             +--+--+--+--+--+--+              | V3 |   V1   |V0|
+ * |          |  |    |        |             |  |  |  |  |  |  |              |    |        |  |
+ * +----------+--+----+--------+-------------++-++-+-++--+--+--+--------------+--^-+-----^--+-^+
+ *            0  5    12                      |  |   |                           |       |    |
+ *                                            +--+---+---------------------------+-------+----+
+ *                                               |   |                           |       |
+ *                                               +---+---------------------------+-------+
+ *                                                   |                           |
+ *                                                   +---------------------------+
+ *
+ *
+ *  The above structure represents the way variable sized keys and values may be
+ *  stored in memory.
+ *  Node Hdr or Node Header will store all the relevant info regarding this node
+ *  type.
+ *  The keys wil be added from the start of the node in an increasing way
+ *  whereas the values will be added from the end of the node in a decreasing 
+ *  way.
```

Comment:
This way the Keys start populating from the left of the node (after the Node Header) while the Values start populating from the right of the node. As the records get added the empty space between the Keys list and the Values list starts to shrink.